### PR TITLE
lib/model: Fix checking children when trying to delete a dir (fixes #6646)

### LINKED ISSUE #6646: Misleading error message when to be deleted dir contains receive-only changes
https://forum.syncthing.net/t/problems-when-syncing-from-windows-to-synology-nas/14996

When a directory deleted on a remote contains receive-only files, the following error message is given:

> directory has been deleted on a remote device but contains unexpected files, scheduling scan

Receive-only files aren't unexpected and rescanning won't help.

### PR DESCRIPTION
**Pull request by imsodin**
https://forum.syncthing.net/t/problems-when-syncing-from-windows-to-synology-nas/14996

When a directory deleted on a remote contains receive-only files, rescanning won't help and the current error message wont direct the user at the actual problem (locally changed files).

Also the scan chan cannot be nil, that might be true in tests, but not everywhere else (we send to it without checking for nil-ness in various other places). And if it is nil in tests, that should be changed instead of having weird code paths like this.

### CODE REVIEW

File: lib/model/folder_sendrecv.go [hunk 1]
```diff
@@ -1825,14 +1824,13 @@ func (f *sendReceiveFolder) deleteDirOnDisk(dir string, snap *db.Snapshot, scanC
 		} else if f.ignores != nil && f.ignores.Match(fullDirFile).IsIgnored() {
 			hasIgnored = true
 		} else if cf, ok := snap.Get(protocol.LocalDeviceID, fullDirFile); !ok || cf.IsDeleted() || cf.IsInvalid() {
+			if f.Type == config.FolderTypeReceiveOnly && cf.IsReceiveOnlyChanged() {
+				hasReceiveOnlyChanged = true
+			}
```
Comment (AudriusButkevicius) [inline lib/model/folder_sendrecv.go:1829]: doesn't this need an else?

**Comment by imsodin** [comment]
While I am addressing one deficiency of that code, I might as well fix it right:  
Just checking `IsInvalid()` makes no sense at this point. What we should check (and now do) is if the file on disk is the same as in db, and if not rescan.
